Yoast/Yoast-SEO-for-TYPO3
# [BUGFIX] Fix support for doctrine/dbal 2.10 by introducing DbalService to use fetch/fetchAssociative when CMS10

## Summary

This PR can be summarized in the following changelog entry:

* CMS10 non-composer ships with `doctrine/dbal` 2.10, which means the `fetchAssociative` and `fetchAllAssociative` methods are not available, this PR fixes that

## Relevant technical choices:

* Introduced `DbalService` which checks if the TYPO3 major version is 10, if so use the old `fetch` and `fetchAll` methods, otherwise the `Associative` methods

## Test instructions

This PR can be tested by following these steps:

* Pull branch and check every functionality if it still works in all versions but especially CMS10

## Quality assurance


Fixes #534
